dask/array/core.py

```diff
@@ -2842,12 +2842,16 @@ def where(condition, x=None, y=None):
     x = broadcast_to(x, shape).astype(dtype)
     y = broadcast_to(y, shape).astype(dtype)
 
-    if isinstance(condition, (bool, np.bool8)):
+    if np.isscalar(condition):
+        condition = bool(condition)
+
+    if isinstance(condition, bool):
         if condition:
```

Comment:
Couldn't this just be...

```python
    if np.isscalar(condition):
        if condition:
        ...
```

Then we skip the casting branch above altogether and let Python do that for us.